source/grid/grid_tools_cache.cc

```diff
@@ -146,6 +152,38 @@ namespace GridTools
       }
     return vertex_kdtree;
   }
+
+
+
+  template <int dim, int spacedim>
+  const RTree<std::pair<BoundingBox<spacedim>, unsigned int>> &
+  Cache<dim, spacedim>::get_covering_rtree() const
+  {
+    if (update_flags & update_covering_rtree)
+      {
+        // TO DO: the locally owned portion of the domain
+        // might consists of two separate pieces: a single
```

Comment:
or more -- I think Burstedde proved that it can't be more than 6 pieces, though :-)